### PR TITLE
[NPUW] Port unroll SDPA optimization from GenAI

### PR DESCRIPTION
**Pull request by AsyaPronina**
### Details:
 - *Copy-pasted Unroll SDPA implementation from GenAI into NPUW ov::npuw::LLMCompiledModel*

### Tickets:
 - *EISW-149347*

### CODE REVIEW

**Comment by dmatveev** [comment]
@AsyaPronina please pick titles carefully.

**Comment by dmatveev** [comment]
@TolyaTalamanov please review

**Comment by dmatveev** [comment]
@TolyaTalamanov gentle reminder

**Comment by dmatveev** [comment]
@TolyaTalamanov gentle reminder

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 1]
```diff
@@ -27,24 +69,200 @@ std::shared_ptr<ov::Model> redirect_new_kv_to_output(const std::shared_ptr<ov::M
...
+            return true;
+        };
+        register_matcher(std::make_shared<opp::Matcher>(matmul, "TransposeValueTensors"), std::move(callback));
+    }
+};
 
-    for (auto tensor : model->inputs()) {
-        if (tensor.get_any_name().find("past_key") != std::string::npos) {
-            ppp.input(tensor.get_any_name()).tensor().set_element_type(ov::element::Type_t::f16);
+class ScaledDotProductAttentionDecomposition : public ov::pass::MatcherPass {
```
Comment (TolyaTalamanov) [inline src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:133]: The pass was initially taken from GPU plugin. Should we re-use already existing pass instead?

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 2]
```diff
@@ -251,41 +469,49 @@ ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& m
     auto kvcache_model = model->clone();
     LOG_DEBUG("2. Transform kvcache model from stateful to stateless.");
     ov::pass::StatefulToStateless().run_on_model(kvcache_model);
-
-    LOG_DEBUG("3. Creating prefill model as clone of transformed kvcache one.");
+    LOG_DEBUG("3. Align u4 ZP constants.");
+    align_u4_zp_constants(kvcache_model);
```
Comment (TolyaTalamanov) [inline src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:473]: I believe it's no longer needed

**Review by TolyaTalamanov — APPROVED** [review]
I'm OK with changes.

Please discuss re-using SDPA unroll pass with GPU plugin team

**Comment by AsyaPronina** [comment]
Ready!

**Pull request merged**